src/components/Photo/DocumentAutoCapture.js

```diff
@@ -93,14 +89,21 @@ export default class DocumentAutoCapture extends Component<Props, State> {
     this.props.onError()
   }
 
+  createWebcamRef = async(ref: React.Ref<typeof Camera>) => {
+    this.webcam = await ref
```

Comment:
So the ref returned from webcamRef is a promise which resolves when the canvas (iirc) initialises? That's much neater, and finally gets rid of the annoying initialisation race condition!

Reply:
Based on the comment below I did some digging and I also don't believe this ref is a promise either? So this would resolve to null?

Reply:
I thought I could async/await without promises (silly me!) and it seemed like this was doing the job, but I'm probably wrong! Maybe I should just use `componentDidUpdate` to check if all the children have updated and then call `start()`. I'm not sure what's the best practice to do this tbh